### ISSUE TITLE
Slug Links

### ISSUE THREAD

**Issue by mastacontrola** [NodeBB/NodeBB#2878]
Filtering through the issues here is quite difficult, so I'm sorry if this issue has already been reported.

Found an issue in the "post_slug_urls" (for lack of better terminology) in that the URL address changes based on the location of the viewport in the browser.

This causes an issue where if you try to go to forum post #3 using a slug URL it may take you to post #5.

Edited for more accurate description.

Basically the slugs that present at the bottom of a post appear to reference the post below it.  However, the link that displays in the drop down appears to reference the proper link.

Hopefully this is definitive enough to find and correct the issue.

**Comment by akhoury** [comment]
+1

a simple example
https://community.nodebb.org/topic/180/who-is-using-nodebb/3

This is scroll to /3 but the url changes to /6. So if I were to share it again, it now scrolls to /6 instead of /3

I think this is happening is because the URL is changing based on the post at the bottom of the viewport where when it auto-scrolls, it sets the target-post to the top of the viewport.

I think the right way to fix this is to behave like the native deep linking behavior of the browser using DOM element IDs, which assume the top of the viewport all the time.

so, for example, in order for the URL to change to /3, the post-n-3 needs to be at the top of the viewport, and when auto-scrolling, the behavior stays the same, the post-n-3 will be at the top of the viewport.

one caveat is that you can never link to the most last post if that post can never hit the top of the viewport. Say post /15. However one should still be able to copy a direct hyperlink with /15 in it, the scroller will do its best to set it at the top of the viewport, if it can't because it hit the limit of the page (and you should not be able to negative-scroll, that's fine this is how the native deep linking works anyways) but the post would still be "highlighted"

**Comment by barisusakli** [comment]
To fix this we need to untie the /index in the url and the index in the navigator (x out of postcount) Right now they use the same value AFAIK.
